### PR TITLE
feat: tree view for scopes

### PR DESCRIPTION
**Pull request by derrabauke**

### CODE REVIEW

File: addon/components/data-table.hbs [hunk 1]
```diff
@@ -34,19 +34,18 @@
...
-         <UkSpinner @ratio="3" />
+          <UkSpinner @ratio="3" />
         </UkFlex>
       </td>
     </tbody>
   {{else}}
     {{yield
       (hash
-        body=(component "data-table/body" models=this.models)
+        body=(component "data-table/body" models=this.data.value)
```
Comment (derrabauke) [inline addon/components/data-table.hbs:44]: As we are using `ember-resources` now, the data fetching is delayed until this component is loaded. This means, using the `<DataTable as |table|/>` without  the `table.body` call would result in a table which never fetches data. Is this valid (the use-case for this scenario seems rare...) or should we lift the fetching artifically higher in the component tree? @czosel

File: addon/controllers/scopes.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Controller from "@ember/controller";
+import { inject as service } from "@ember/service";
+
+export default class ScopesController extends Controller {
+  @service router;
+
+  get activeScope() {
+    if (!this.router.currentRouteName.includes("ember-emeis.scopes.edit")) {
```
Comment (derrabauke) [inline addon/controllers/scopes.js:8]: this feels not right. What do you think @czosel ?

File: tests/integration/components/tree-test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      level: 1,
+      parent: root,
+    });
+    const grandchildren = this.server.createList("scope", 2, {
+      level: 2,
+      parent: level1[0],
+    });
+
+    grandchildren.forEach((grandchild) => {
+      grandchild.findParents = () => {
```
Comment (derrabauke) [inline tests/integration/components/tree-test.js:24]: Could we lift this somehow into the factory?

File: addon/controllers/scopes.js (hunk 2 above)
Comment (czosel) [reply addon/controllers/scopes.js:8]: @velrest and I just took another look at this. We didn't find a good alternative, but we did find some documentation for `attributes`: https://emberjs.github.io/rfcs/0095-router-service.html#routeinfowithattributes
I'd stay let's stick with it :slightly_smiling_face:

File: addon/components/data-table.hbs (hunk 1 above)
Comment (czosel) [reply addon/components/data-table.hbs:44]: I agree, data-tables without body don't seem to make sense.

File: tests/integration/components/tree-test.js (hunk 3 above)
Comment (czosel) [reply tests/integration/components/tree-test.js]: Fixed :upside_down_face:

**Review by czosel — APPROVED** [review]
:clap:

**Pull request merged**